cmd/treedb_text_hybrid_scale/main.go

```
@@ -1467,13 +1795,31 @@ func writeReports(rep report) error {
-	if err := os.WriteFile(rep.Artifacts.Markdown, []byte(renderMarkdown(rep)), 0o644); err != nil {
+	return atomicWriteFile(rep.Artifacts.Markdown, []byte(renderMarkdown(rep)))
+}
+
+func atomicWriteFile(path string, data []byte) error {
+	tmp, err := os.CreateTemp(filepath.Dir(path), "."+filepath.Base(path)+"-*")
+	if err != nil {
 		return err
 	}
-	return nil
+	tmpName := tmp.Name()
+	defer os.Remove(tmpName)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)